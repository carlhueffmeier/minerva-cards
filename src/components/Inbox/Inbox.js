import React, { Component } from 'react';
import Button from 'components/Button';
import NewDeck from 'components/NewDeck';
import { Link } from 'react-router-dom';

export default class Inbox extends Component {
  goToReview() {
    this.props.history.push(`/review`);
  }

  render() {
    return (
      <div>
        <h1>Inbox</h1>
        <ul className="deck-list">
          {this.props.decks.map(deck => (
            <li key={deck.id}>
              <Link to={`/decks/${deck.id}`}>{deck.title}</Link>
            </li>
          ))}
        </ul>

        <NewDeck />

        <Button onClick={this.goToReview.bind(this)}>Start Learning</Button>
      </div>
    );
  }
}
