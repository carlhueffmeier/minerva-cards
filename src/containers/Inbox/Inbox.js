import React from 'react';
import { Button } from 'components';

function goToReview() {
  console.log('changing route');
}

function Inbox(props) {
  return (
    <div className="App">
      <h1>Inbox</h1>
      <p>Filter by: </p>
      <ul>
        <li>
          <a href="#">Due Decks</a>
        </li>
        <li>
          <a href="#">Active Decks</a>
        </li>
        <li>
          <a href="#">Inactive Decks</a>
        </li>
      </ul>
      <ul>
        <li>
          <a href="#">JavaScript</a>
        </li>
        <li>
          <a href="#">Functional Programming</a>
        </li>
      </ul>

      <Button onClick={goToReview}>Start Learning</Button>
    </div>
  );
}

export default Inbox;
