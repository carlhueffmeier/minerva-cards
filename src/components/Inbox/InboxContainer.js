import { connect } from 'react-redux';
import { selectors as deckSelectors } from 'state/modules/decks';
import Inbox from './Inbox';

function mapStateToProps(state, ownProps) {
  return {
    decks: deckSelectors.getAllDecks(state)
  };
}

export default connect(mapStateToProps, null)(Inbox);
