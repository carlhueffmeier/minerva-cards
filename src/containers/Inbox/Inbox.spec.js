import React from 'react';
import { shallow } from 'enzyme';
import Inbox from './Inbox';

describe('Inbox', () => {
  const inbox = shallow(<Inbox />);

  it('renders properly', () => {
    expect(inbox).toMatchSnapshot();
  });
});
