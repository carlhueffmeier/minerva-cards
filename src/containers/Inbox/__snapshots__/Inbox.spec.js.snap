// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inbox renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inbox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Inbox
</h1>,
        <p>
          Filter by: 
</p>,
        <ul>
          <li>
                    <a
                              href="#"
                    >
                              Due Decks
                    </a>
          </li>
          <li>
                    <a
                              href="#"
                    >
                              Active Decks
                    </a>
          </li>
          <li>
                    <a
                              href="#"
                    >
                              Inactive Decks
                    </a>
          </li>
</ul>,
        <ul>
          <li>
                    <a
                              href="#"
                    >
                              JavaScript
                    </a>
          </li>
          <li>
                    <a
                              href="#"
                    >
                              Functional Programming
                    </a>
          </li>
</ul>,
        <Button
          onClick={[Function]}
>
          Start Learning
</Button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Inbox",
        },
        "ref": null,
        "rendered": "Inbox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Filter by: ",
        },
        "ref": null,
        "rendered": "Filter by: ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a
                            href="#"
              >
                            Due Decks
              </a>
</li>,
            <li>
              <a
                            href="#"
              >
                            Active Decks
              </a>
</li>,
            <li>
              <a
                            href="#"
              >
                            Inactive Decks
              </a>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                Due Decks
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Due Decks",
                "href": "#",
              },
              "ref": null,
              "rendered": "Due Decks",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                Active Decks
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Active Decks",
                "href": "#",
              },
              "ref": null,
              "rendered": "Active Decks",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                Inactive Decks
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Inactive Decks",
                "href": "#",
              },
              "ref": null,
              "rendered": "Inactive Decks",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a
                            href="#"
              >
                            JavaScript
              </a>
</li>,
            <li>
              <a
                            href="#"
              >
                            Functional Programming
              </a>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                JavaScript
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "JavaScript",
                "href": "#",
              },
              "ref": null,
              "rendered": "JavaScript",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                Functional Programming
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Functional Programming",
                "href": "#",
              },
              "ref": null,
              "rendered": "Functional Programming",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Start Learning",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Start Learning",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Inbox
</h1>,
          <p>
            Filter by: 
</p>,
          <ul>
            <li>
                        <a
                                    href="#"
                        >
                                    Due Decks
                        </a>
            </li>
            <li>
                        <a
                                    href="#"
                        >
                                    Active Decks
                        </a>
            </li>
            <li>
                        <a
                                    href="#"
                        >
                                    Inactive Decks
                        </a>
            </li>
</ul>,
          <ul>
            <li>
                        <a
                                    href="#"
                        >
                                    JavaScript
                        </a>
            </li>
            <li>
                        <a
                                    href="#"
                        >
                                    Functional Programming
                        </a>
            </li>
</ul>,
          <Button
            onClick={[Function]}
>
            Start Learning
</Button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inbox",
          },
          "ref": null,
          "rendered": "Inbox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Filter by: ",
          },
          "ref": null,
          "rendered": "Filter by: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a
                                href="#"
                >
                                Due Decks
                </a>
</li>,
              <li>
                <a
                                href="#"
                >
                                Active Decks
                </a>
</li>,
              <li>
                <a
                                href="#"
                >
                                Inactive Decks
                </a>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  Due Decks
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Due Decks",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Due Decks",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  Active Decks
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Active Decks",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Active Decks",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  Inactive Decks
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Inactive Decks",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Inactive Decks",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a
                                href="#"
                >
                                JavaScript
                </a>
</li>,
              <li>
                <a
                                href="#"
                >
                                Functional Programming
                </a>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  JavaScript
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "JavaScript",
                  "href": "#",
                },
                "ref": null,
                "rendered": "JavaScript",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  Functional Programming
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Functional Programming",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Functional Programming",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Start Learning",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Start Learning",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
